tests/test_project_contibutor_views.py

```diff
@@ -0,0 +1,569 @@
+        assert self.project.has_permission(self.user1, permissions.READ) is True
+
+    def test_contributor_manage_reorder(self):
+
+        # Two users are added as a contributor via a POST request
+        project = ProjectFactory(creator=self.user1, is_public=True)
+        reg_user1, reg_user2 = UserFactory(), UserFactory()
+        project.add_contributors(
+            [
+                {'user': reg_user1, 'permissions': permissions.ADMIN, 'visible': True},
+                {'user': reg_user2, 'permissions': permissions.ADMIN, 'visible': False},
+            ]
+        )
+        # Add a non-registered user
+        unregistered_user = project.add_unregistered_contributor(
+            fullname=fake.name(), email=fake_email(),
+            auth=self.consolidate_auth1,
+            save=True,
+        )
+
+        url = project.api_url + 'contributors/manage/'
+        self.app.post(
+            url,
+            json={
+                'contributors': [
+                    {
+                        'id': reg_user2._id,
+                        'permission': permissions.ADMIN,
+                        'registered': True,
+                        'visible': False
+                    },
+                    {
+                        'id': project.creator._id,
+                        'permission': permissions.ADMIN,
+                        'registered': True,
+                        'visible': True
+                    },
+                    {
+                        'id': unregistered_user._id,
+                        'permission': permissions.ADMIN,
+                         'registered': False,
+                        'visible': True
+                    },
+                    {
+                        'id': reg_user1._id,
+                        'permission': permissions.ADMIN,
+                        'registered': True,
+                        'visible': True
+                    },
+                ]
+            },
+            auth=self.auth,
+        )
+
+        project.reload()
+
+        # Note: Cast ForeignList to list for comparison
+        assert list(project.contributors) == [reg_user2, project.creator, unregistered_user, reg_user1]
+
+        assert list(project.visible_contributors) == [project.creator, unregistered_user, reg_user1]
+
+    def test_project_remove_contributor(self):
+        # User 1 removes user2
+        self.app.post(
+            self.project.api_url_for('project_remove_contributor'),
+            json={
+                'contributorID': self.user2._id,
+                'nodeIDs': [
+                    self.project._id
+                ]
+            },
+            auth=self.auth,
+            follow_redirects=True
+        )
+        self.project.reload()
+        assert self.user2._id not in self.project.contributors
+        # A log event was added
+        assert self.project.logs.latest().action == 'contributor_removed'
+
+    def test_project_remove_curator(self):
```

Comment:
This is the location for the new test to be added, as you can see it's was difficult to locate where in all these related view tests we were checking for log behavior that is why I am splitting up all these test files.